src/Microsoft.AspNet.OData.Shared/DefaultEdmPatchMethodHandler.cs

```diff
@@ -0,0 +1,170 @@
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// This is the default patch handler for non-CLR types. This class has default get, create and update
+    /// methods that are used to patch an original collection when the collection is provided.
+    /// </summary>
+    internal class DefaultEdmODataAPIHandler : EdmODataAPIHandler
+    {
+        private IEdmEntityType entityType;
+        private ICollection<IEdmStructuredObject> originalList;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultEdmODataAPIHandler"/> class.
+        /// </summary>
+        /// <param name="originalList">Original collection of the type which needs to be updated.</param>
+        /// <param name="entityType">The Edm entity type of the collection.</param>
+        public DefaultEdmODataAPIHandler(ICollection<IEdmStructuredObject> originalList, IEdmEntityType entityType)
+        {
+            Debug.Assert(entityType != null, "entityType != null");
+
+            this.entityType = entityType;
+            this.originalList = originalList ?? new List<IEdmStructuredObject>();
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryGet(IDictionary<string, object> keyValues, out IEdmStructuredObject originalObject, out string errorMessage)
+        {
+            ODataAPIResponseStatus status = ODataAPIResponseStatus.Success;
+            errorMessage = string.Empty;
+            originalObject = null;
+
+            Debug.Assert(keyValues != null, "keyValues != null");
+
+            try
+            {
+                originalObject = GetFilteredItem(keyValues);
+
+                if (originalObject == null)
+                {
+                    status = ODataAPIResponseStatus.NotFound;
+                }
+            }
+            catch (Exception ex)
+            {
+                status = ODataAPIResponseStatus.Failure;
+                errorMessage = ex.Message;
+            }
+
+            return status;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryCreate(IDictionary<string, object> keyValues, out IEdmStructuredObject createdObject, out string errorMessage)
+        {
+            createdObject = null;
+            errorMessage = string.Empty;
+
+            try
+            {
+                createdObject = new EdmEntityObject(entityType);
```

Comment:
Can the `EdmEntityObject` constructor used here throw an exception? Is the `try...catch` here serving any purpose?

Reply:
`originalList.Add(createdObject);` can cause an exception.